extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxCoreRecorder.java

```diff
@@ -318,18 +322,33 @@ public Integer get() {
 
     static class VertxSupplier implements Supplier<Vertx> {
         final VertxConfiguration config;
+        final VertxOptionsCustomizer customizer;
         Vertx v;
 
-        VertxSupplier(VertxConfiguration config) {
+        VertxSupplier(VertxConfiguration config, List<Consumer<VertxOptions>> customizers) {
             this.config = config;
+            this.customizer = new VertxOptionsCustomizer(customizers);
         }
 
         @Override
         public synchronized Vertx get() {
             if (v == null) {
-                v = initialize(config);
+                v = initialize(config, customizer);
             }
             return v;
         }
     }
+
+    static class VertxOptionsCustomizer {
+        final List<Consumer<VertxOptions>> customizers;
+
+        VertxOptionsCustomizer(List<Consumer<VertxOptions>> customizers) {
+            this.customizers = customizers;
+        }
+
+        VertxOptions customize(VertxOptions options) {
+            customizers.forEach(x -> x.accept(options));
```

Comment:
We try and avoid lambdas in runtime code, they are slower than normal for loops and use a lot more metaspace. It's not much taken individually, but if you have hundreds of places that are lambdas rather than for loops it uses significantly more memory.

We are less strict about deployment code, as it does not end up in the final application.

Reply:
Removed/undone. =)